core/watcher.ts

```diff
@@ -77,30 +77,31 @@ export default class FSWatcher implements Watcher {
   async start() {
     const { root, ignore, debounce } = this.options;
     const watcher = Deno.watchFs(root);
-    const changes = new Set<string>();
+    const changeQueue: Set<string>[] = [];
     let timer = 0;
     let runningCallback = false;
 
     await this.dispatchEvent({ type: "start" });
 
     const callback = async () => {
-      if (!changes.size || runningCallback) {
-        return;
-      }
-
-      const files = new Set(changes);
-      changes.clear();
-
       runningCallback = true;
 
-      try {
-        const result = await this.dispatchEvent({ type: "change", files });
-        if (false === result) {
-          return watcher.close();
+      let changes: Set<string> | undefined;
+      while ((changes = changeQueue.pop()) !== undefined) {
+        try {
+          const result = await this.dispatchEvent({
```

Comment:
Instead of dispatching an event per change in the queue, why don't merge all changes and dispatch the event only once? I mean:

```js
let changes: Set<string> | undefined;
let mergedChanges = new Set<string>();

while ((changes = changeQueue.pop()) !== undefined) {
  mergedChanges = mergedChanges.union(changes);
}
```

Reply:
Right, it didn't even occur to me that we could batch all future changes into the same one. This is a great idea!